File: backend/app/routers/disputes.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession
from app.core.dependencies import get_current_active_user
from app.core.database import get_db
from app.services.dispute_service import DisputeService

router = APIRouter()

@router.post("/")
async def raise_dispute(
    booking_id: int,
    dispute_type: str,
    description: str,
    current_user = Depends(get_current_active_user),
    db: AsyncSession = Depends(get_db)
):
    service = DisputeService(db)
    return await service.create_dispute(booking_id, current_user.id, dispute_type, description)

@router.get("/{dispute_id}")
async def get_dispute(dispute_id: int, current_user = Depends(get_current_active_user), db: AsyncSession = Depends(get_db)):
    service = DisputeService(db)
    return await service.get_dispute(dispute_id, current_user)

@router.post("/{dispute_id}/messages")
async def send_dispute_message(
    dispute_id: int,
    message: str,
    current_user = Depends(get_current_active_user),
    db: AsyncSession = Depends(get_db)
):
    service = DisputeService(db)
    return await service.add_message(dispute_id, current_user.id, "user", message)

File: frontend/handlers/disputes.js
// src/handlers/disputes.js
import api from "../api/api";

// ─── DISPUTE HANDLERS ─────────────────────────────────────────────────────────

/**
 * Raise a new dispute
 * POST /disputes/
 */
export const raiseDispute = async (bookingId, disputeType, description) => {
  const { data } = await api.post("/disputes/", null, {
    params: {
      booking_id: bookingId,
      dispute_type: disputeType,
      description,
    },
  });
  return data;
};

/**
 * Get dispute details
 * GET /disputes/{dispute_id}
 */
export const fetchDispute = async (disputeId) => {
  const { data } = await api.get(`/disputes/${disputeId}`);
  return data;
};

/**
 * Send message in dispute
 * POST /disputes/{dispute_id}/messages
 */
export const sendDisputeMessage = async (disputeId, message) => {
  const { data } = await api.post(`/disputes/${disputeId}/messages`, null, {
    params: { message },
  });
  return data;
};

/**
 * Get dispute messages
 * GET /disputes/{dispute_id}/messages
 */
export const fetchDisputeMessages = async (disputeId) => {
  const { data } = await api.get(`/disputes/${disputeId}/messages`);
  return data;
};